activesupport/test/inflector_test.rb

```diff
@@ -140,10 +140,26 @@ def test_camelize
     end
   end
 
-  def test_camelize_with_lower_downcases_the_first_letter
+  def test_camelize_with_true_upcases_the_first_letter
+    assert_equal("Capital", ActiveSupport::Inflector.camelize("Capital", true))
+  end
+
+  def test_camelize_with_upper_upcases_the_first_letter
+    assert_equal("Capital", ActiveSupport::Inflector.camelize("Capital", :upper))
+  end
+
+  def test_camelize_with_false_downcases_the_first_letter
     assert_equal("capital", ActiveSupport::Inflector.camelize("Capital", false))
   end
 
+  def test_camelize_with_lower_downcases_the_first_letter
+    assert_equal("capital", ActiveSupport::Inflector.camelize("Capital", :lower))
+  end
+
+  def test_camelize_with_any_other_arg_downcases_the_first_letter
+    assert_equal("capital", ActiveSupport::Inflector.camelize("Capital", 222))
```

Comment:
Is there any reason to change the behavior from upper to lower?

Preserving the original behavior is more safer.

Or if we intend to match `String#camelize`, how about raising an ArgumentError the same with `String#camelize`?

https://github.com/rails/rails/blob/291a3d2ef29a3842d1156ada7526f4ee60dd2b59/activesupport/lib/active_support/core_ext/string/inflections.rb#L103-L112

Reply:
I think the original behavior was a bug, or at least, was surprising. The intent is to match `String#camelize`.

I don't mind switching to an error if this causes problems but I prefer to have the framework handle more cases correctly if it can.

Reply:
If an user was doing `ActiveSupport::Inflector.camelize("Capital", :true)` in app, this change will silently flip the behavior from upper to lower.

Is this the user's responsibility? IMO no.

Reply:
Raised a PR https://github.com/rails/rails/pull/41317.

Reply:
Ahh yep I get you now. How about this? https://github.com/rails/rails/pull/41318